Azure-Samples/active-directory-dotnetcore-daemon-v2
# Updated first sample to use dependency injection and included unit tests

## Purpose

* Use dependency injection pattern
* Use `Options` pattern to deal with `appsettings.json` configuration instead of original `AuthenticationConfig.cs` class
* Include sample unit tests with mocks

## Does this introduce a breaking change?

```
[ ] Yes
[x] No
```

## Pull Request Type
What kind of change does this Pull Request introduce?


```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Documentation content changes
[ ] Other... Please describe:
```

## How to Test
*  Get the code
* Test the code. Should work same as before.

```
cd daemon-console
dotnet run
```
* Run unit tests (do this from the same directory as the `daemon-console.sln` file).

```
dotnet test
```


## What to Check
Verify that the following are valid
* Tests run and pass
* Code runs the same as before

## Other Information


This is meant to serve as a template for possible paths forward with respect to testing and even new patterns in other applications that leverage `ASP.NET Core`. Also makes it possible to leverage library components through injectable services.
